storage/src/test/java/tech/pegasys/teku/storage/server/kvstore/AbstractDatabaseTest.java

```diff
@@ -1211,33 +1211,36 @@ public void shouldThrowIfClosedDatabaseIsRead_getHistoricalState() throws Except
   @Test
   public void shouldThrowIfTransactionModifiedAfterDatabaseIsClosedFromAnotherThread()
       throws Exception {
+    database.storeInitialAnchor(genesisAnchor);
+    final StampedLock lock = new StampedLock();
+    final long stamp = lock.writeLock();
+    final Thread dbCloserThread =
+        new Thread(
+            () -> {
+              try {
+                database.close();
+              } catch (Exception e) {
+                throw new RuntimeException(e);
+              } finally {
+                lock.unlock(stamp);
+              }
+            });
 
-    for (int i = 0; i < 20; i++) {
```

Comment:
I think the loop here was deliberate since this can't be entirely deterministic.